features/expo/deviceInfo/hooks.ts

```diff
@@ -0,0 +1,82 @@
+import { useEffect, useState } from 'react';
+
+import { enumToStr } from '../../../utils/enum';
+import * as Device from 'expo-device';
+import { DateTime } from 'luxon';
+
+import { DeviceInfo } from './constants';
+import { DeviceType } from './enums';
+
+export const useDeviceInfo = () => {
+    const [deviceInfo, setDeviceInfo] = useState<DeviceInfo>({
+        brand: null,
+        manufacturer: null,
+        deviceName: null,
+        modelId: null,
+        modelName: null,
+        osName: null,
+        osVersion: null,
+        osBuildId: null,
+        osInternalBuildId: null,
+        deviceType: null,
+        deviceYearClass: null,
+        isDevice: null,
+        supportedCpuArchitectures: null,
+        totalMemory: null,
+        uptime: null,
+        isJailBroken: null,
+    });
+
+    useEffect(() => {
+        const fetchDeviceInfo = async () => {
```

Comment:
Incorrect. Please use `.then()` structure.

```ts
Device.getDeviceTypeAsync().then(deviceType => {
    setDeviceInfo(prev => ({...prev, deviceType: enumToStr(DeviceType, deviceType})
})).catch(error => {
    console.error('Failed to fetch device type:', error);
});
```

Please apply to the other async functions as well.